app/controllers/ProjectSidewalkAPIController.scala

```diff
@@ -164,34 +177,114 @@ class ProjectSidewalkAPIController @Inject()(implicit val env: Environment[User,
+          val averagedStreetFeatures = streetAccessScores.map(_.attributes).transpose.map(_.sum / streetAccessScores.size).toArray
+          val accessScore: Double = computeAccessScore(averagedStreetFeatures, significance)
 
+          val allStreetsIntersectingTheNeighborhood = allStreetEdges.filter(_.geom.intersects(neighborhood.geom))
+          val coverage: Double = auditedStreetsIntersectingTheNeighborhood.size.toDouble / allStreetsIntersectingTheNeighborhood.size
+
+          assert(coverage <= 1.0)
+
+          writer.println(neighborhood.name.getOrElse("NA") + "," + neighborhood.regionId + "," + accessScore + "," + coordinatesToString + "," + coverage + "," + averagedStreetFeatures(0) + "," + averagedStreetFeatures(1) + "," + averagedStreetFeatures(2) + "," + averagedStreetFeatures(3) + "," + 
+                        significance(0) + "," + significance(1) + "," + significance(2) + "," + significance(3))                
+        } else {
+          writer.println(neighborhood.name.getOrElse("NA") + "," + neighborhood.regionId + "," + None.asInstanceOf[Option[Double]] + "," + coordinatesToString + ",,,,,," + significance(0) + "," + significance(1) + "," + significance(2) + "," + significance(3))
+        }
+      }
+      writer.close()
+      file
+    }
+    prepareFeatureCollectionCSV
+  }
+
+  /**
+    *
+    * returns a list of AttributeForAccessScore to reduce redundancy between methods
+    *
+    * @param version
+    * @param coordinates
+    * @param return
+    */
+  def getLabelsForScore(version: Int, coordinates: Array[Double]): List[AttributeForAccessScore] = {
+    val labelsForScore: List[AttributeForAccessScore] = version match {
+      case 1 =>
+        val labelLocations: List[LabelLocation] = LabelTable.selectLocationsOfLabelsIn(coordinates(0), coordinates(2), coordinates(1), coordinates(3))
+        val clusteredLabelLocations: List[LabelLocation] = clusterLabelLocations(labelLocations)
+        clusteredLabelLocations.map(l => AttributeForAccessScore(l.lat, l.lng, l.labelType))
+      case 2 =>
+        val globalAttributes: List[GlobalAttributeForAPI] = GlobalAttributeTable.getGlobalAttributesInBoundingBox(coordinates(0).toFloat, coordinates(2).toFloat, coordinates(1).toFloat, coordinates(3).toFloat, None)
+        globalAttributes.map(l => AttributeForAccessScore(l.lat, l.lng, l.labelType))
+    }
+    labelsForScore
+  }
+
+  /**
+    *
+    * generic version of getAccessScoreNeighborhood, makes changes for v1 vs v2, for GeoJSON file format only
+    *
+    * @param lat1
+    * @param lng1
+    * @param lat2
+    * @param lng2
+    * @param version
+    * @param requestStr
+    * @param coordinates
+    * @return
+    */
+  def getAccessScoreNeighborhoodsGeneric(lat1: Double, lng1: Double, lat2: Double, lng2: Double, version: Int, requestStr: String, coordinates: Array[Double]): JsObject = {
```

Comment:
Can you rename this function `getAccessScoreNeighborhoodsJson` and rename the other one `getAccessScoreNeighborhoodsCSV`?